### ISSUE TITLE
Needrestart flaw

### ISSUE THREAD

**Issue by alithechemist** [mail-in-a-box/mailinabox#2461]
Five local privilege escalation (LPE) vulnerabilities have been discovered in the needrestart utility used by Ubuntu Linux, which was introduced over 10 years ago in version 21.04.

The flaws were discovered by Qualys and are tracked as CVE-2024-48990, CVE-2024-48991, CVE-2024-48992, CVE-2024-10224, and CVE-2024-11003. They were introduced in needrestart version 0.8, released in April 2014, and fixed only yesterday, in version 3.8.

I just checked and my instance is running needrestart 3.5
[https://www.bleepingcomputer.com/news/security/ubuntu-linux-impacted-by-decade-old-needrestart-flaw-that-gives-root/]

```
root@ma:~# needrestart --version

needrestart 3.5 - Restart daemons after library updates.

Authors:
  Thomas Liske <thomas@fiasko-nw.net>

Copyright Holder:
  2013 - 2020 (C) Thomas Liske [http://fiasko-nw.net/~thomas/]

Upstream:
  https://github.com/liske/needrestart

This program is free software; you can redistribute it and/or modify
it under the terms of the GNU General Public License as published by
the Free Software Foundation; either version 2 of the License, or
(at your option) any later version.

```

Thanks,
Chip

**Comment by matidau** [comment]
Did you check your updates?

3.5-5ubuntu2.2 is the updated version I think.

**Comment by matidau** [comment]
The advisory states that the affected versions are
<= 3.5-5ubuntu2.1

So the version you have installed 3.5-5ubuntu2.2 
is the updated version without the exploit.

**Comment by alithechemist** [comment]
I see that it's >3.8.  I'm gonna close this because I replaced it with a dummy script.
I'll put it back when canonical releases an update, I'm aware of the consequences.

> They were introduced in needrestart version 0.8, released in April 2014, and fixed only yesterday, in version 3.8.

Chip
